scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py

```diff
@@ -0,0 +1,304 @@
+        self.horizontal_layout.addItem(self.spacer_item)
+        self.horizontal_layout.setAlignment(QtCore.Qt.AlignLeft)
+
+        self.vertical_layout = QtGui.QVBoxLayout(self)
+        self.vertical_layout.setObjectName("verticalLayout")
+        self.vertical_layout.addWidget(self.grouping_table)
+        self.vertical_layout.addLayout(self.horizontal_layout)
+
+        self.setLayout(self.vertical_layout)
+
+    def set_up_table(self):
+        self.grouping_table.setColumnCount(3)
+        self.grouping_table.setHorizontalHeaderLabels(["Group Name", "Detector IDs", "N Detectors"])
+        header = self.grouping_table.horizontalHeader()
+        header.setResizeMode(0, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(1, QtGui.QHeaderView.Stretch)
+        header.setResizeMode(2, QtGui.QHeaderView.ResizeToContents)
+        vertical_headers = self.grouping_table.verticalHeader()
+        vertical_headers.setMovable(False)
+        vertical_headers.setResizeMode(QtGui.QHeaderView.ResizeToContents)
+        vertical_headers.setVisible(True)
+
+        self.grouping_table.horizontalHeaderItem(0).setToolTip("The name of the group :"
+                                                               "\n    - The name must be unique across all groups/pairs"
+                                                               "\n    - The name can only use digits, characters and _")
+        self.grouping_table.horizontalHeaderItem(1).setToolTip("The sorted list of detectors :"
+                                                               "\n  - The list can only contain integers."
+                                                               "\n  - , is used to separate detectors or ranges."
+                                                               "\n  - \"-\" denotes a range, i,e \"1-5\" is the same as"
+                                                               " \"1,2,3,4,5\" ")
+        self.grouping_table.horizontalHeaderItem(2).setToolTip("The number of detectors in the group.")
+
+    def num_rows(self):
+        return self.grouping_table.rowCount()
+
+    def num_cols(self):
+        return self.grouping_table.columnCount()
+
+    def notify_data_changed(self):
+        if not self._updating:
+            self.dataChanged.emit()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Adding / removing table entries
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def add_entry_to_table(self, row_entries):
+        assert len(row_entries) == self.grouping_table.columnCount()
+
+        row_position = self.grouping_table.rowCount()
+        self.grouping_table.insertRow(row_position)
+        for i, entry in enumerate(row_entries):
+            item = QtGui.QTableWidgetItem(entry)
+            if group_table_columns[i] == 'group_name':
```

Comment:
should use the `group_table_columns` at the top of the file

Reply:
Yes